### PR TITLE
DOC: improve the vonmises docstring example

### LINKED ISSUE #5811: numpy.random.vonmises documentation contains incorrect code example.
The numpy.random [reference](http://docs.scipy.org/doc/numpy/reference/generated/numpy.random.vonmises.html) for the von Mises distribution contains an incorrect code sample (pasted below).

```python
mu, kappa = 0.0, 4.0 # mean and dispersion
s = np.random.vonmises(mu, kappa, 1000)

import matplotlib.pyplot as plt
import scipy.special as sps
count, bins, ignored = plt.hist(s, 50, normed=True)
x = np.arange(-np.pi, np.pi, 2*np.pi/50.)
y = -np.exp(kappa*np.cos(x-mu))/(2*np.pi*sps.jn(0,kappa))
plt.plot(x, y/max(y), linewidth=2, color='r')
plt.show()
```

While this works for the example, line 5 doesn't concur with the definition of the von Mises probability density function, given above the example. According to documentation about the [special functions] (http://docs.scipy.org/doc/scipy/reference/special.html#module-scipy.special), the modified bessel function of order real order is given by scipy.special.iv(). Using that function instead means the sign doesn't need to be changed.

I noticed the problem when I tried to emulate the code and I found that area under the pdf didn't sum to 1 when I used the trapezium rule and would sometimes even go below the x axis, when neither the exponential function or the bessel function should ever be negative. Compare:

```python
#incorrect, according to documentation

import numpy as np
import scipy.special as sps
import matplotlib.pyplot as plt
tau = np.pi*2

fig, ax = plt.subplots(1,1, figsize = (10,8))

n = 10
xs = np.linspace(0, 4, n)     #create a range of kappas

mu = 0
m = 50
width = tau/float(m)
sums = np.empty(n)    #for storing the integrals under the curves for different kappas

for i in range(n):
    kappa = xs[i]
    x = np.linspace(-np.pi, np.pi, m)
    y = -np.exp(kappa*np.cos(x-mu))/(tau*sps.jn(0,kappa))
    ax.plot(x, y, label="kappa = {}".format(kappa))
    sums[i] = sum([(y[j]+y[j+1])/2*width for j in range(m-1)])
ax.set_axes("tight")
ax.legend()
print "the area under the pdf for different kappas: "
for i in range(n):
    print "kappa =", xs[i], ", integral =", sums[i]
fig.show()
```

```python
#correct, using the correct bessel function

import numpy as np
import scipy.special as sps
import matplotlib.pyplot as plt
tau = np.pi*2

fig, ax = plt.subplots(1,1, figsize = (10,8))

n = 10
xs = np.linspace(0, 4, n)     #create a range of kappas

mu = 0
m = 50
width = tau/float(m)
sums = np.empty(n)    #for storing the integrals under the curves for different kappas

for i in range(n):
    kappa = xs[i]
    x = np.linspace(-np.pi, np.pi, m)
    y = np.exp(kappa*np.cos(x-mu))/(tau*sps.iv(0,kappa))
    ax.plot(x, y, label="kappa = {}".format(kappa))
    sums[i] = sum([(y[j]+y[j+1])/2*width for j in range(m-1)])
ax.set_axes("tight")
ax.legend()
print "the area under the pdf for different kappas: "
for i in range(n):
    print "kappa =", xs[i], ", integral =", sums[i]
fig.show()
```

Excuse my use of tau.

Using: Python 2.7.9 :: Anaconda 2.2.0 (64-bit), IPython 3.1.0, numpy 1.9.2

### PR DESCRIPTION
**Pull request by argriffing**
closes https://github.com/numpy/numpy/issues/5811

### CODE REVIEW

File: numpy/random/mtrand/mtrand.pyx [hunk 1]
```diff
@@ -2524,11 +2524,11 @@ cdef class RandomState:
...
-        >>> import scipy.special as sps
-        >>> count, bins, ignored = plt.hist(s, 50, normed=True)
-        >>> x = np.arange(-np.pi, np.pi, 2*np.pi/50.)
-        >>> y = -np.exp(kappa*np.cos(x-mu))/(2*np.pi*sps.jn(0,kappa))
-        >>> plt.plot(x, y/max(y), linewidth=2, color='r')
+        >>> from scipy.special import i0
+        >>> plt.hist(s, 50, normed=True)
+        >>> x = np.linspace(-np.pi, np.pi, num=51)
+        >>> y = np.exp(kappa*np.cos(x-mu))/(2*np.pi*i0(kappa))
+        >>> plt.plot(x, y, linewidth=2, color='r')
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Tested? I note that `y` is no longer normalized.

Comment (argriffing) [inline numpy/random/mtrand/mtrand.pyx]: Right, `y` is not divided by its maximum value anymore.  But why were they doing that originally?  I think that with this change, both the histogram and pdf should integrate to ~1.0.

Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: I'm guessing that the curve was supposed to pass through the tops of the histogram bars. Might be worth running a check...

Comment (argriffing) [inline numpy/random/mtrand/mtrand.pyx]: No, the `normed` argument of [`hist`](http://matplotlib.org/api/pyplot_api.html#matplotlib.pyplot.hist) normalizes the area not the maximum.  So in http://docs.scipy.org/doc/numpy/reference/generated/numpy.random.vonmises.html the histogram has area 1.0 but the density curve has maximum 1.0, and it's mostly a coincidence that the maximum of the histogram is also near 1.0. If not for this coincidence, the problems mentioned in https://github.com/numpy/numpy/issues/5811 probably would have been found sooner.

Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: I'm not arguing at this point ;) I'd just like to see the plot...

**Comment by argriffing** [comment]
> I'm not arguing at this point ;) I'd just like to see the plot...

```python
import numpy as np

mu, kappa = 0.0, 4.0 # mean and dispersion
s = np.random.vonmises(mu, kappa, 1000)

# Display the histogram of the samples, along with
# the probability density function:
                                 
import matplotlib.pyplot as plt
from scipy.special import i0
plt.hist(s, 50, normed=True)
x = np.linspace(-np.pi, np.pi, num=51)
y = np.exp(kappa*np.cos(x-mu))/(2*np.pi*i0(kappa))
plt.plot(x, y, linewidth=2, color='r')
plt.savefig('vonmises.png')
```
vonmises

This depends on the random number seed, and it uses the same number of samples and bins as before. Increasing the number of samples will make the histogram agree more closely with the curve.

**Comment by charris** [comment]
Great, thanks @argriffing .

**Pull request merged**